GDCpp/GDCpp/Extensions/Builtin/RuntimeSceneTools.cpp

```diff
@@ -208,6 +211,48 @@ bool GD_API PickNearestObject(std::map <gd::String, std::vector<RuntimeObject*>
     return true;
 }
 
+bool GD_API RaycastObject(std::map <gd::String, std::vector<RuntimeObject*> *> pickedObjectLists, float x, float y, float angle, float dist, gd::Variable & varX, gd::Variable & varY, bool inverted)
+{
+    RuntimeObject * matchObject = NULL;
+    float testSqDist = inverted ? 0 : dist*dist;
+    float resultX = 0.0f;
+    float resultY = 0.0f;
+    for (auto it = pickedObjectLists.begin(); it != pickedObjectLists.end(); ++it)
+    {
+        if ( it->second == NULL ) continue;
+        auto list = *it->second;
+
+        for (std::size_t i = 0; i < list.size(); ++i)
+        {
+
+            RaycastResult result = list[i]->RaycastTest(x, y, angle, dist, !inverted);

```

Comment:
I'm thinking about an optimisation (not sure if it is worth implementing but could be useful):

Before doing the RaycastTest, which can be costly I guess, we could check the bounding radius of the object, and the distance between the ray source position and the object position minus the bouding radius.
If this value is greater thant the testSqDist, then we don't even need to do the RaycastTest as if the result is a collision, it will be farther than the current result.

What do you think? (Let me know if it's not clear)

Reply:
Well, maybe it's over-optimization because we specify a maximum distance so this should skip quite a few objects from the full test.

Reply:
And this would also make the code more complex... the only way to know would be to do benchmarks. Not a big deal for now.

Reply:
The bounding check is already done in the RuntimeObject::RaycastTest method, it's a shame that a sqrt() has to be used, but it's the only way :/

Reply:
I see!